packages/visual-editor/src/components/Dropzone/Hitboxes.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { CSSProperties, useCallback, useEffect, useMemo, useState } from 'react';
+import styles from './styles.module.css';
+import { useTreeStore } from '@/store/tree';
+import { getItemDepthFromTree } from '@/utils/getItem';
+import { CTFL_ZONE_ID, DRAGGABLE_WIDTH, ROOT_ID } from '@/types/constants';
+import { useZoneStore } from '@/store/zone';
+import { useDraggedItemStore } from '@/store/draggedItem';
+import { createPortal } from 'react-dom';
+
+interface Props {
+  parentZoneId: string;
+  zoneId: string;
+  enableRootHitboxes: boolean;
+}
+
+const INITIAL_OFFSET = 10;
+const OFFSET_INCREMENT = 8;
+const HITBOX_HEIGHT = 20;
+const HITBOX_WIDTH = 80;
+const MIN_HEIGHT = 45;
+const MIN_DEPTH = 20;
+enum HitboxDirection {
+  TOP,
+  LEFT,
+  RIGHT,
+  BOTTOM,
+  SELF_VERTICAL,
+  SELF_HORIZONTAL,
+}
+
+const calcOffsetDepth = (depth: number) => {
+  return INITIAL_OFFSET - OFFSET_INCREMENT * depth;
+};
+
+const Hitboxes: React.FC<Props> = ({ zoneId, parentZoneId, enableRootHitboxes }) => {
+  const tree = useTreeStore((state) => state.tree);
+  const isDraggingOnCanvas = useDraggedItemStore((state) => state.isDraggingOnCanvas);
+  const zoneDepth = useMemo(
+    () => getItemDepthFromTree({ id: parentZoneId }, tree),
+    [tree, parentZoneId],
+  );
+  const [fetchDomRect, setFetchDomRect] = useState(Date.now());
+
+  useEffect(() => {
```

Comment:
You could try useLayoutEffect - I forget the details but I think this is a case where it's worth looking into. Basically the useLayoutEffect hook is used to paint the css and make style related dom edits.

Reply:
Ticket for this: https://contentful.atlassian.net/browse/ALT-542